source/shadow.c

```diff
@@ -372,27 +376,50 @@ static ShadowStatus_t containsSubString( const char * pString,
 
 static ShadowStatus_t validateName( const char * pString,
                                     uint16_t stringLength,
+                                    uint16_t maxAllowedLength,
                                     uint16_t * pNameLength )
 {
     uint16_t index = 0U;
     ShadowStatus_t returnStatus = SHADOW_FAIL;
+    uint8_t parsedName = 0U;
```

Comment:
nit: don't we have an `enum` or `#define` for boolean values?

Reply:
We don't. We will be able to create one, but it will be for this one case and the readability will not see a major jump with that change in my opinion. Do you think it will be better to create an enum in this case?